tests/testthat/test-determine_relevant_years.R

```diff
@@ -0,0 +1,9 @@
+test_that("returns appropriate length integer vector", {
+  market_share_target_reference_year <- 2023
+  time_horizon <- 5
+  output <- determine_relevant_years(market_share_target_reference_year, time_horizon)
+  expect_vector(output, ptype = integer())
+  expect_length(output, time_horizon + 1)
+  expect_contains(output, market_share_target_reference_year)
+  expect_length(output, length(unique(output)))
```

Comment:
For this one, I think having an explicit `expect_identical` is a good idea, since these checks aren't looking for continuity, or max/min value (having a `2040` sneak in somehow).

```suggestion
  expect_length(output, length(unique(output)))
  expect_identical(output, c(2023L, 2024L, 2025L, 2026L, 2027L, 2028L))
```

Reply:
since the input values are parameterized, I'd rather it be
```suggestion
  expect_length(output, length(unique(output)))
  expect_identical(output, market_share_target_reference_year: (market_share_target_reference_year + time_horizon))
```
but that's basically replicating the function 🤣 
https://github.com/RMI-PACTA/pacta.data.preparation/blob/a43cd61e36379e74a5334735944fd9ac80cd1a77/R/determine_relevant_years.R#L15

Reply:
But the input values aren't parameterized in the test... That's the point of the test.

Reply:
```r
expect_identical(
  determine_relevant_years(market_share_target_reference_year, time_horizon),
  determine_relevant_years(market_share_target_reference_year, time_horizon)
)
```
🚀